thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java

```diff
@@ -362,10 +362,6 @@ public Response sendEmailWithText(
       throw new WebApplicationException("Empty : list of recipients" + toAddr);
     }
 
-    if (Strings.isNullOrEmpty(smtpHost)) {
-      smtpHost = thirdeyeConfiguration.getSmtpHost();
-    }
-
     SmtpConfiguration smtpConfiguration = getSmtpConfiguration(smtpHost, smtpPort);
     if (smtpConfiguration == null) {
```

Comment:
The return of getSmtpConfiguration() cannot be null. You should add a checker in the function, and return null if the smtpConfiguration isn't valid.

Reply:
Thanks for pointing this out. Changed in next commit.